scrape/scrape_test.go

```diff
@@ -1138,6 +1138,125 @@ func TestTargetScrapeScrapeNotFound(t *testing.T) {
 	}
 }
 
+func TestMutateSampleLabels(t *testing.T) {
+
+	tests := []struct {
+		Title      string
+		ScrapePool *scrapePool
+		Input      struct {
+			Lset   labels.Labels
+			Target *Target
+		}
+		Expected labels.Labels
+	}{
+		{
+			Title: "Exported",
+			ScrapePool: &scrapePool{
+				config: &config.ScrapeConfig{
+					HonorLabels: false,
+				},
+			},
+			Input: struct {
+				Lset   labels.Labels
+				Target *Target
+			}{
+				Lset: []labels.Label{
+					{
+						Name:  "n",
+						Value: "v1",
+					},
+				},
+				Target: &Target{
+					labels: []labels.Label{
+						{
+							Name:  "n",
+							Value: "v2",
+						},
+					},
+				},
+			},
+			Expected: []labels.Label{
+				{
+					Name:  "exported_n",
+					Value: "v1",
+				},
+				{
+					Name:  "n",
+					Value: "v2",
+				},
+			},
+		}, {
+			Title: "Honor Labels",
+			ScrapePool: &scrapePool{
+				config: &config.ScrapeConfig{
+					HonorLabels: true,
+				},
+			},
+			Input: struct {
+				Lset   labels.Labels
+				Target *Target
+			}{
+				Lset: []labels.Label{
+					{
+						Name:  "n",
+						Value: "v",
+					},
+				},
+				Target: &Target{
+					labels: []labels.Label{
+						{
+							Name:  "n",
+							Value: "v0",
+						},
+						{
+							Name:  "n1",
+							Value: "v1",
+						},
```

Comment:
I'd inline those like `{Name: "n1", Value: "v1"}` so the lines don't bloat up as bad. I find it hard to read a single test unit otherwise.